flowapi/flowapi/user_model.py

```diff
@@ -0,0 +1,285 @@
+
+        Parameters
+        ----------
+        action: {'run', 'poll', 'get_results'}
+            Action to check
+        query_kind : str
+            Kind of the query
+        aggregation_unit : str
+            Aggregation unit/level of resolution
+
+        Returns
+        -------
+        bool
+            True if the user can do 'action' with this query
+
+        Raises
+        ------
+        UserClaimsVerificationError
+            If the user cannot do action with this kind of query at this level of aggregation
+        """
+        try:
+            action_rights = self.claims[query_kind]["permissions"][action]
+            unit_rights = (
+                aggregation_unit in self.claims[query_kind]["spatial_aggregation"]
+            )
+            if not action_rights:
+                raise UserClaimsVerificationError(
+                    f"Token does not allow {action} for query kind '{query_kind}'"
+                )
+            if not unit_rights:
+                raise UserClaimsVerificationError(
+                    f"Token does not allow query kind '{query_kind}' at spatial aggregation '{aggregation_unit}'"
+                )
+        except KeyError:
+            raise UserClaimsVerificationError("Claims verification failed.")
+        return True
+
+    def can_run(self, *, query_kind: str, aggregation_unit: str) -> bool:
+        """
+        Returns true if the user can run this kind of query at this unit of aggregation.
+
+        Parameters
+        ----------
+        query_kind : str
+            Kind of the query
+        aggregation_unit : str
+            Aggregation unit/level of resolution
+
+        Returns
+        -------
+        bool
+            True if the user can run this query
+
+        Raises
+        ------
+        UserClaimsVerificationError
+            If the user cannot run this kind of query at this level of aggregation
+
+        """
+
+        return self.has_access(
+            action="run", query_kind=query_kind, aggregation_unit=aggregation_unit
+        )
+
+    async def can_poll_by_query_id(self, *, query_id) -> bool:
+        """
+        Returns true if the user can poll this query.
+
+        Parameters
+        ----------
+        query_id : str
+            Identifier of the query.
+
+        Returns
+        -------
+        bool
+            True if the user can get the status of this query
+
+        Raises
+        ------
+        UserClaimsVerificationError
+            If the user cannot get the status of this kind of query at this level of aggregation
+        """
+        request.socket.send_json(
+            {
+                "request_id": request.request_id,
+                "action": "get_query_params",
+                "params": {"query_id": query_id},
+            }
+        )
+        reply = await request.socket.recv_json()
+        if reply["status"] == "success":
```

Comment:
What happens if `reply["status"]` is `"error"`? Shouldn't this raise an exception as well? (And similarly below?)